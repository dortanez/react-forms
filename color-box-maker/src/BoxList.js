import React, {useState} from 'react';
import Box from './Box';
import NewBoxForm from './NewBoxForm';
import {v4 as uuid} from 'uuid';
import './BoxList.css';

const BoxList = () => {
    const [boxes, setBoxes] = useState([]);
    const createBox = newBox => {
        setBoxes(boxes => 
            [...boxes,{...newBox, key: uuid()}])
    }
    const deleteBox = id => {
        setBoxes(boxes => boxes.filter(box => box.id !== id))
    }
    return (
        <div>
            <NewBoxForm createBox={createBox}/>
            <div className='BoxList-boxes'>
                {boxes.map(({height, width, color, id}) => (
                    <Box 
                        deleteBox={deleteBox} 
                        height={height} 
                        width={width} 
                        color={color} 
                        key={id} 
                        id={id}
                    />
                ))}
            </div>
        </div>
    )
}

export default BoxList;