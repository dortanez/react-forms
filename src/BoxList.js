import React from 'react';
import Box from './Box';
import NewBoxForm from './NewBoxForm';

const BoxList = () => {
    return (
        <div>

        </div>
    )
}

export default BoxList;