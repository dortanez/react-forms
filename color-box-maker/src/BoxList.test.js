import { render, fireEvent } from '@testing-library/react';
import BoxList from './BoxList';

const addBox = (boxList, height = "100", width = "100", color = "green") => {
    const heightInput = boxList.getByLabelText("Height");
    const widthInput = boxList.getByLabelText("Width");
    const backgroundInput = boxList.getByLabelText("Color");
    fireEvent.change(backgroundInput, { target: { value: color } });
    fireEvent.change(widthInput, { target: { value: width } });
    fireEvent.change(heightInput, { target: { value: height } });
    const button = boxList.getByText("Submit");
    fireEvent.click(button);
  }

it('should render without crashing',() => {
    render(<BoxList/>)
})

it('should match snapshot',() => {
    const {asFragment} = render(<BoxList/>);
    expect(asFragment()).toMatchSnapshot();
})

it('can create box',() => {
    const boxList = render(<BoxList/>);
    expect(boxList.queryByText('X')).not.toBeInTheDocument();
    addBox(boxList);
    expect(boxList.queryByText('X')).toBeInTheDocument();
})

it('can delete box',() => { 
    const boxList = render(<BoxList/>);
    addBox(boxList);
    const removeButton = boxList.getByText('X');
    fireEvent.click(removeButton);
    expect(removeButton).not.toBeInTheDocument();
})