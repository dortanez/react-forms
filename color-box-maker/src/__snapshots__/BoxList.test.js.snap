// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <h4>
        Create New Box
      </h4>
      <label
        for="color"
      >
        Color
      </label>
      <input
        id="color"
        name="color"
        type="color"
        value="#000000"
      />
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        name="height"
        type="number"
        value="0"
      />
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        name="width"
        type="number"
        value="0"
      />
      <button>
        Submit
      </button>
    </form>
    <div
      class="BoxList-boxes"
    />
  </div>
</DocumentFragment>
`;
